### ISSUE TITLE
[indent] incorrect indentation expectation for type parameters

### ISSUE THREAD

**Issue by mgummelt** [typescript-eslint/typescript-eslint#3764]
- [ X] I have tried restarting my IDE and the issue persists.
- [ X] I have updated to the latest version of the packages.
- [ X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "indent": "off",
    "@typescript-eslint/indent": "error"
  }
}
```

```TS
const result =
    await this.client.query<
          GetTablesByNameResponse,
          TablesByNamesVars>({
                query: GET_TABLES_BY_NAME,
                variables: { databaseID, names },
          });
```

**Expected Result**

No lint error.

**Actual Result**

"Expected Indentation of 12 spaces, but found 16".

ESlint is expecting the type parameters `GetTablesByNameResponse` and `TablesByNamesVars` to be aligned with `await`, which I don't think is correct, since they belong to the same line of code, and should therefore be indented.  VSCode's default typescript formatter tries to indent them, for example.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.29.2` |
| `@typescript-eslint/parser`        | `4.29.2` |
| `TypeScript`                       | `4.3.5` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `14.17.5` |

**Comment by bradzacher** [comment]
#1824 please use the search

**Comment by mgummelt** [comment]
That's a pretty generic issue to discover via search, but alright.

**Comment by bradzacher** [comment]
It's one of two issues about the indent rule that are open, and it's the second result that shows up when you search for "indent":

![image](https://user-images.githubusercontent.com/7462525/130271421-589f85ed-1984-49bc-a5d1-41f2e76fe48c.png)
